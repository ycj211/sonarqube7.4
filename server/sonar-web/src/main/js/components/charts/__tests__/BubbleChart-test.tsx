/*
 * SonarQube
 * Copyright (C) 2009-2018 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import * as React from 'react';
import { mount } from 'enzyme';
import BubbleChart, { Bubble } from '../BubbleChart';

it('should display bubbles', () => {
  const items = [{ x: 1, y: 10, size: 7 }, { x: 2, y: 30, size: 5 }];
  const chart = mount(<BubbleChart height={100} items={items} />);
  chart.find(Bubble).forEach(bubble => expect(bubble).toMatchSnapshot());
});

it('should render bubble links', () => {
  const items = [{ x: 1, y: 10, size: 7, link: 'foo' }, { x: 2, y: 30, size: 5, link: 'bar' }];
  const chart = mount(<BubbleChart height={100} items={items} />);
  chart.find(Bubble).forEach(bubble => expect(bubble).toMatchSnapshot());
});

it('should render bubbles with click handlers', () => {
  const onClick = jest.fn();
  const items = [{ x: 1, y: 10, size: 7, data: 'foo' }, { x: 2, y: 30, size: 5, data: 'bar' }];
  const chart = mount(<BubbleChart height={100} items={items} onBubbleClick={onClick} />);
  chart.find(Bubble).forEach(bubble => expect(bubble).toMatchSnapshot());
});
