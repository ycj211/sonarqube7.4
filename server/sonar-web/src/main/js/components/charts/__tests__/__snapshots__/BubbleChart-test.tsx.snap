// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display bubbles 1`] = `
<Bubble
  key="0"
  r={45}
  scale={1}
  x={-10}
  y={52.3015873015873}
>
  <Tooltip>
    <g>
      <circle
        className="bubble-chart-bubble"
        r={45}
        style={
          Object {
            "fill": undefined,
            "stroke": undefined,
          }
        }
        transform="translate(-10, 52.3015873015873) scale(1)"
      />
    </g>
  </Tooltip>
</Bubble>
`;

exports[`should display bubbles 2`] = `
<Bubble
  key="1"
  r={33.57142857142857}
  scale={1}
  x={-75}
  y={33.57142857142857}
>
  <Tooltip>
    <g>
      <circle
        className="bubble-chart-bubble"
        r={33.57142857142857}
        style={
          Object {
            "fill": undefined,
            "stroke": undefined,
          }
        }
        transform="translate(-75, 33.57142857142857) scale(1)"
      />
    </g>
  </Tooltip>
</Bubble>
`;

exports[`should render bubble links 1`] = `
<Bubble
  key="0"
  link="foo"
  r={45}
  scale={1}
  x={-10}
  y={52.3015873015873}
>
  <Tooltip>
    <g>
      <Link
        onlyActiveOnIndex={false}
        style={Object {}}
        to="foo"
      >
        <a
          onClick={[Function]}
          style={Object {}}
        >
          <circle
            className="bubble-chart-bubble"
            r={45}
            style={
              Object {
                "fill": undefined,
                "stroke": undefined,
              }
            }
            transform="translate(-10, 52.3015873015873) scale(1)"
          />
        </a>
      </Link>
    </g>
  </Tooltip>
</Bubble>
`;

exports[`should render bubble links 2`] = `
<Bubble
  key="1"
  link="bar"
  r={33.57142857142857}
  scale={1}
  x={-75}
  y={33.57142857142857}
>
  <Tooltip>
    <g>
      <Link
        onlyActiveOnIndex={false}
        style={Object {}}
        to="bar"
      >
        <a
          onClick={[Function]}
          style={Object {}}
        >
          <circle
            className="bubble-chart-bubble"
            r={33.57142857142857}
            style={
              Object {
                "fill": undefined,
                "stroke": undefined,
              }
            }
            transform="translate(-75, 33.57142857142857) scale(1)"
          />
        </a>
      </Link>
    </g>
  </Tooltip>
</Bubble>
`;

exports[`should render bubbles with click handlers 1`] = `
<Bubble
  data="foo"
  key="0"
  onClick={[MockFunction]}
  r={45}
  scale={1}
  x={-10}
  y={52.3015873015873}
>
  <Tooltip>
    <g>
      <circle
        className="bubble-chart-bubble"
        onClick={[Function]}
        r={45}
        style={
          Object {
            "fill": undefined,
            "stroke": undefined,
          }
        }
        transform="translate(-10, 52.3015873015873) scale(1)"
      />
    </g>
  </Tooltip>
</Bubble>
`;

exports[`should render bubbles with click handlers 2`] = `
<Bubble
  data="bar"
  key="1"
  onClick={[MockFunction]}
  r={33.57142857142857}
  scale={1}
  x={-75}
  y={33.57142857142857}
>
  <Tooltip>
    <g>
      <circle
        className="bubble-chart-bubble"
        onClick={[Function]}
        r={33.57142857142857}
        style={
          Object {
            "fill": undefined,
            "stroke": undefined,
          }
        }
        transform="translate(-75, 33.57142857142857) scale(1)"
      />
    </g>
  </Tooltip>
</Bubble>
`;
